compiler/src/middle_end/optimize_dead_assignments.re

```diff
@@ -14,21 +17,61 @@ let safe_to_remove_import = i =>
   | _ => false
   };
 
+let is_used = ident =>
+  Option.is_some(Ident.find_same_opt(ident, used_symbols^));
```

Comment:
We should add some utilities to the Ident module (like `contains`). Probably not needed in this PR.